lib/kalibro_client/entities/processor/metric_collector_details.rb

```diff
@@ -33,8 +33,7 @@ def find_metric_by_name(name)
         end
 
         def find_metric_by_code(metric_code)
-          metric = self.supported_metrics.find {|code, metric| metric.code == metric_code}
-          metric.nil? ? nil : metric.last
+          @supported_metrics[metric_code]
```

Comment:
Sure, but find_by_metric_name is a concern right now?

Em qui, 7 de mai de 2015 às 13:42, pedroscocco <notifications@github.com>
escreveu:

> In lib/kalibro_client/entities/processor/metric_collector_details.rb
> <https://github.com/mezuro/kalibro_client/pull/43#discussion_r29870546>:
>
> > @@ -33,8 +33,7 @@ def find_metric_by_name(name)
> >          end
> >
> >          def find_metric_by_code(metric_code)
> > -          metric = self.supported_metrics.find {|code, metric| metric.code == metric_code}
> > -          metric.nil? ? nil : metric.last
> > +          @supported_metrics[metric_code]
>
> Could we create another hash supported_metrics_by_name to be used in
> find_metric_by_name?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mezuro/kalibro_client/pull/43/files#r29870546>.
>